### ISSUE TITLE
Brainstorm subscription prompts across the site (design)

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#3924]
We have a great subscription system now. Let's come up with some better, smarter, and smoother ways to encourage people to subscribe (without just making BIG SIGNS EVERYWHERE). Ideas:

* [ ] combined “Subscribe and login” flow - get people subscribed to some tags as they sign up, based on tags from the pages they've been looking at
    * a route or parameters or session modification to pass some tag names in upon logging in, which then are auto-subscribed to with a notification and link to /subscriptions
* [ ] combined “Comment + subscribe” flow - get people subscribed to some tags as they leave a comment, maybe based on the tags of the post they're commenting on, perhaps using parameters to the comment post action

What other ideas do you have? Feel free to post sketches or brainstorm here!

**Comment by jywarren** [comment]
@SidharthBansal just jotting some of these down here! 👍

**Comment by SidharthBansal** [comment]
Great

**Comment by SidharthBansal** [comment]
In our current system for every signup/login comment, we want the following for the backend:
1) Model for selection of tags where I person comments/subscribe and to find related tags
2)Controller action to handle the different basic and alternative flows:
    i)Basic Flows
Person comments and is logged in through OAuth provider Facebook
Person comments  and is logged in through OAuth provider Twitter
Person comments is logged in through OAuth provider Github
Person comments and is logged in through OAuth provider Google
Person comments  and is logged in through the basic sign up form
Person comments and is signed in through OAuth provider Facebook
Person comments and is signed in through OAuth provider Twitter
Person comments and is signed in through OAuth provider Github
Person comments is signed in through OAuth provider Google
Person comments and is signed in through the basic sign up form
I listed in this fashion so that we will not forget any of these basic flows. Note that the terms written above differ significantly. Log in and sign up are different, etc. 
   ii) Alternative Flows: Note that the article means research note/wiki page. We have tags for both pages.
A person has commented on the article and then he has signed up through the basic form.
A person has commented on the article and then he has logged in through the basic form.
A person has commented on the article and then he has logged in through the Facebook
A person has commented on the article and then he has signed up through the Facebook
A person has commented on the article and then he has logged in through the Twitter
A person has commented on the article and then he has signed up through the Twitter
A person has commented on the article and then he has signed up through the Google
A person has commented on the article and then he has logged in through the Google
A person has commented on the article and then he has signed up through the Github
A person has commented on the article and then he has logged in through the Github
A person has commented on the article which does not have any tags. So what we need to subscribe to? We need to brainstorm there also.
The behaviour will be checked on the local, unstable and production branch. 
3) Testing: We need to test for comments feature for all the four providers and for the basic form for both login/signup process.
So this will require around 10 tests. We will also require some additional integration tests. You taught me TDD so well Jeff. Thanks for it. I will never forget the importance of tests.
4) Frontend:I don't think, here we do require any piece of front end.

The above list is valid if we are making subscriptions on the basis of the tags already present in the note/wiki page. We also need to brainstorm the process in absence of tags on the page or what other similar tags we can get the user subscribed.

>>combined “Subscribe and login” flow - get people subscribed to some tags as they sign up, based on tags from the pages they've been looking at

This will be somewhat similar to the above.

**Comment by SidharthBansal** [comment]
It is informal list. @jywarren please edit the things in the above comments itself so that the conversation is short in this issue.
I am planning to do ``combined “Subscribe and login” flow`` first. Then we can do the other parts. What do you say?
Also, it will be better if we can divide the issue into two sub issues. Just creating 2 issues and linking back here.

**Comment by jywarren** [comment]
This sounds great, thanks @SidharthBansal. 

A new idea too: 

* [ ] "Suggested subscriptions" on your dashboard based on current subscriptions + "related tags" (see this on tag pages like https://publiclab.org/tag/infragram)
* [ ] "Suggested subscriptions" on email notification footers

**Comment by SidharthBansal** [comment]
We can brainstorm all the ideas in this issue. For the issues which needs implementation we will create separate issues and try to solve them. This way we will have this issue reserved for the mainly discussion.

I would also like to suggest
Wiki page + subscription on like
Research note + subscribe on like
More soon

**Comment by SidharthBansal** [comment]
Can we use trending tags as given here
https://github.com/publiclab/plots2/blob/232a77f7e1e43aa37a9cab1436fef1eec34fab2d/app/models/tag.rb#L268
 to suggest tags to be subscribed on these pages?

**Comment by SidharthBansal** [comment]
>> Both trending and related tags could be relevant to use here.


Do we have a relevant tag function? I can't find it.

**Comment by jywarren** [comment]
we have tag.related and a Tag.trending functions, but not tag.relevant... what would that do? Thanks!

**Comment by jywarren** [comment]
How about when you like something, it pops up something saying "Liked! Follow more content like this by clicking one of these topics: ...."

**Comment by SidharthBansal** [comment]
We can do this:
Case A1. User likes the page
A2. We will show buttons under each tag "Liked! Follow more content like this by clicking one of these topics: ....".
Case B. A user has not liked the page. Then no need to show subscribe buttons under each tag.

**Comment by SidharthBansal** [comment]
We forgot the main area where the suggestions to subscribe should be.
https://publiclab.org/subscriptions
We can show popular/trending tags on this webpage on side bar along with icons to subscribe.
What do you say?
@jywarren you can :+1:  on the comments which you wish should be implemented. I will create separate issues for them.
For those which are not needed to be implemented please do :-1: .
For others please comment.
This willl create the conversation short.

**Comment by SidharthBansal** [comment]
Email's footer for the subscriptions.
We can create a Milestone for this? What do you say?

**Comment by jywarren** [comment]
Yes sorry adding 👍 now. Re /subscriptions I think that page is nice but it's more of a utility page and less a page which people will notice regularly like /dashboard.

**Comment by jywarren** [comment]
Feel free to organize ideas into a list here or elsewhere, however you'd like to structure it! Also there are some folks who are getting good at making new issues who've been in the chatroom recently so it's a good place to ask for help in making break-out issues if you'd like.

**Comment by jywarren** [comment]
Awesome, @SidharthBansal !

**Comment by jywarren** [comment]
Let's make a break-out issue for including tags to subscribe to in a login sequence, just like the `return_to` works, but with `subscribe=tag,tag` so that we can make a variety of ways to display buttons that allow signing up and subscribing or logging in and subscribing. It can be a general-use route that lets us build lots of different prompts as we've listed above (for ppl who are not yet signed up or logged in). Make sense? Thanks!!

**Comment by SidharthBansal** [comment]
@jywarren. I have holidays for the whole month. Exams over!!! :smile: I will be doing the work as quickly as I can. 

https://github.com/publiclab/plots2/issues/4131 is the formal list of issues. We will discuss issues here and paste the main title there with checkbox.
I will create the issues for all the checkboxes after we complete https://github.com/publiclab/plots2/issues/3924#issuecomment-444536503 work and some login modal work.

**Comment by jywarren** [comment]
:-)))

haha take a break first! lol

hope exams went really well. I'm sure it was a lot of work.

Great, and congrats!

**Comment by SidharthBansal** [comment]
>>hope exams went really well. I'm sure it was a lot of work.

Yeah a lot of stuff. :-)
